src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -397,8 +402,69 @@ def makeSubscriptions(self, workflow):
+                            if workflow['isRelVal']:
+                                pnnExpr = '('+'|'.join(["pnn=%s" % pnn for pnn in dMap['destination']])+')'
+                                rseExpression = pnnExpr + '&cms_type=real&rse_type=DISK'
+                            else:
+                                rseExpression = '(tier=2|tier=1)&cms_type=real&rse_type=DISK'
+                            scope = 'cms'
+                            if self.msConfig['enableDataPlacement']:
+                                rucioResultList.append(self.rucio.createReplicationRule(dMap['datasets'],
+                                                                                        rseExpression,
+                                                                                        copies=copies,
+                                                                                        scope=scope))
+                            else:
+                                msg = "Dry run mode. The effective Rucio submission would look like: \n"
+                                msg += "account: %s \n"
+                                msg += "dids: %s \n"
+                                msg += "rseExpression: %s\n"
+                                msg += "copies: %s\n"
+                                self.logger.warning(msg,
+                                                    self.msConfig['rucioAccount'],
+                                                    pformat([{'scope': scope, 'name': did} for did in dMap['datasets']]),
+                                                    rseExpression,
+                                                    copies)
+                        except Exception as ex:
+                            msg = "Could not make transfer subscription for Workflow: %s\n:%s"
+                            self.logger.exception(msg, workflow['RequestName'], str(ex))
+                            return workflow
+                else:
+                    # NOTE:
+                    #    Nothing else to be done here. We mark the document as
+                    #    done so we do not iterate through it multiple times
+                    msg = "Skip submissions for %s. Either all data Tiers were "
+                    msg += "excluded or there were no Output Datasets at all."
+                    self.logger.warning(msg, workflow['RequestName'])
+                    self.docKeyUpdate(workflow, transferStatus='done')
+                    return workflow
+
+                transferIDs = rucioResultList
+                self.docKeyUpdate(workflow,
+                                  transferStatus='done',
+                                  transferIDs=transferIDs)
+                return workflow
+
+            elif isinstance(workflow, (list, set, CommandCursor)):
+                # FIXME
+                pass
+            else:
+                msg = "Unsupported type %s for workflows!\n" % type(workflow)
```

Comment:
I think this will not help too much, because it is not related to the workflow type, but rather to destingush the way/mode in which the `makeTransferSubscription` method is called - either for a single workflow or for a set of workflows. This line here should never be hit at  runtime. If we end up here this is a definite sign for an error in the code. I would rather think of rising an exception here. Your opinion is quite valuable here, so please do share it.

Reply:
That's a good idea! Raise an exception with a clear message, such that placement for this workflow fails - without affecting the rest of the workflows in the pipeline.

Reply:
And I checked, but rising an exception is actually a bad idea - it breaks the pipeline and needs to be handled outside - which leads to not needed complications. As I previously said this, is not expected to change during runtime with some workflows or any change of running conditions. This is to distinguish the actual run mode - how  the pipeline/thread is feeded

Reply:
I see. I think we should log the content of `workflow` then, in addition to its type. Unless it's already done in the previous blocks of code.

Reply:
I wouldn't touch this piece here. Trying to log something you do not know what it is or how long it is, is not a good idea. Especially when you know it is a clear signal that the method is misused. This should stay as a clear message to the programmer that the call is wrong. It will not help us logging workflow contents, because in this situation the worklflow won't be relevant at all (and it will be broken for any workflow), it may just create obscure message and distract the one who is looking at the logs from the important fact - this is a signal that the method is called in a mode it is not designed to work. Or even worse trying to log something of an unknown type may break it in unexpected way and again miss the important message.

Reply:
Ok! Lets break the whole module in case we run into this one. Mybe there is some point of fully attracting the attention of the developers like that. I implemented a `UnsupportedError` exception, similarly to the `NotImplementedError` one.

Reply:
Done